### DISCUSSION TITLE
Add readonly mode

### DISCUSSION THREAD

**Discussion by tryoxiss** [helix-editor/helix#9245] [General] [answered]
Hi, I was kinda shocked to find out that helix dosen't have a proper read only mode--given its mentioned multiple times. Or at least its not documented anywhere aside from the option to show `[READONLY]` in the statusline (which is the only mention of it in the docs at all).

I found [an action execution](https://github.com/helix-editor/helix/actions/runs/5066297244/workflow), but using the flag resuklts in a could not parse arguments error. (Helix 23.10). I read the entire (very short) help page, which had nothing, did multiple searches in the docs and outside of them.

Is it just a planned thing and thsts poorly communicated? Am I missing something?

**Comment by kirawi** [comment, marked as answer]
Do you mean opening a file in readonly mode even if you have permissions to write? That is not supported (see https://github.com/helix-editor/helix/pull/7128#issuecomment-1581560237). It is only opened readonly if the file is readonly.

**Reply by tryoxiss** [reply]
Yeah pretty much! Just feels weird that basically every other editor has this but helix is missing it -- since its very useful for when you want just a bit more than a pager (e.g., read a code file with syntax hilighting). Even just a `--read` flag (like is mentioned the [#7128 comment](https://github.com/helix-editor/helix/pull/7128#issuecomment-1581560237) in which makes it behave as if you didnt have write permissions would be incredible!

**Comment by matthias-t** [comment]
I sometimes use helix as a pager with `cat FILE | hx`. This does not automatically work with syntax highlighting, though, but one could use `:set-language`.
